File: 21-Greedy-Algorithms/02-Interval-Scheduling/cpp/non_overlapping.cpp
/* Non-overlapping Intervals | LeetCode 435 | Tags: interval-scheduling, greedy */
#include <bits/stdc++.h>
using namespace std;
int eraseOverlapIntervals(vector<vector<int>>& intervals){sort(intervals.begin(),intervals.end(),[](auto&a,auto&b){return a[1]<b[1];});
int cnt=0,end=INT_MIN;for(auto&i:intervals){if(i[0]>=end)end=i[1];else cnt++;}return cnt;}
int main(){vector<vector<int>>intervals={{1,2},{2,3},{3,4},{1,3}};cout<<eraseOverlapIntervals(intervals)<<endl;} // 1

File: 21-Greedy-Algorithms/02-Interval-Scheduling/java/NonOverlapping.java
import java.util.*;
public class NonOverlapping { public static int eraseOverlapIntervals(int[][] intervals){Arrays.sort(intervals,(a,b)->a[1]-b[1]);
    int cnt=0,end=Integer.MIN_VALUE;for(int[] i:intervals){if(i[0]>=end)end=i[1];else cnt++;}return cnt;}
    public static void main(String[] args){System.out.println(eraseOverlapIntervals(new int[][]{{1,2},{2,3},{3,4},{1,3}}));}}
